### PR TITLE
Fix PHP 7.1 related failures

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

PHP 7.1 triggers notices on previously silent implicit type changes.

### CODE REVIEW

**Comment by fabpot** [comment]
Do we need to add 7.1 to the travis matrix then?

File: src/Symfony/Component/DependencyInjection/composer.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
         "php": ">=5.3.9"
     },
     "require-dev": {
-        "symfony/yaml": "~2.1",
+        "symfony/yaml": "~2.3.42|~2.7.14|~2.8.7|~3.0.7|~3.1.1|~3.2",
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/composer.json]: #18861 required on PHP 7.1

File: src/Symfony/Component/DependencyInjection/composer.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
         "php": ">=5.3.9"
     },
     "require-dev": {
-        "symfony/yaml": "~2.1",
+        "symfony/yaml": "~2.3.42|~2.7.14|~2.8.7",
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/composer.json]: should be `~2.8.7|~3.0.7|~3.1.1|~3.2` when merging in 3.0

File: src/Symfony/Bridge/Twig/composer.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
         "symfony/security-acl": "~2.6",
         "symfony/stopwatch": "~2.2",
         "symfony/console": "~2.7",
-        "symfony/var-dumper": "~2.6",
+        "symfony/var-dumper": "~2.7.16|~2.8.9",
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/composer.json]: should be `~2.8.9|~3.0.9|~3.1.3|~3.2` when merging in 3.0

File: .travis.yml [hunk 4]
```diff
@@ -50,13 +51,12 @@ before_install:
     - if [[ ! $skip ]]; then echo session.gc_probability = 0 >> $INI_FILE; fi
     - if [[ ! $skip && $PHP = 5.* ]]; then echo extension = mongo.so >> $INI_FILE; fi
     - if [[ ! $skip && $PHP = 5.* ]]; then echo extension = memcache.so >> $INI_FILE; fi
-    - if [[ ! $skip && $PHP = 5.* ]]; then (echo yes | pecl install -f apcu-4.0.10 && echo apc.enable_cli = 1 >> $INI_FILE); fi
-    - if [[ ! $skip && $PHP = 7.* ]]; then (echo yes | pecl install -f apcu-5.1.2 && echo apc.enable_cli = 1 >> $INI_FILE); fi
+    - if [[ ! $skip && $PHP = 5.* ]]; then (echo yes | pecl install -f apcu-4.0.11 && echo apc.enable_cli = 1 >> $INI_FILE); fi
+    - if [[ ! $skip && $PHP = 7.0 ]]; then (echo yes | pecl install -f apcu-5.1.5 && echo apc.enable_cli = 1 >> $INI_FILE); fi
```
Comment (nicolas-grekas) [inline .travis.yml]: apcu on 7.1 segfaults for now

**Comment by nicolas-grekas** [comment]
Green, with php7.1 included in the test matrix

File: .travis.yml (hunk 4 above)
Comment (stof) [inline .travis.yml]: have you reported it ?

**Comment by stof** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/DependencyInjection/composer.json (hunk 2 above)
Comment (fabpot) [inline src/Symfony/Component/DependencyInjection/composer.json]: 3.0.9 and 3.1.3, right?

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/composer.json]: 3.0.7&3.1.1 here

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/composer.json]: See #18861

**Pull request merged**